### PR TITLE
Add hapi support for local assets

### PR DESCRIPTION
**Pull request by gingermusketeer**
TODO:
~~-[ ] Test with hapi app and ensure that it works as expected.~~
- [x] Make hapi a dev dependency

### CODE REVIEW

**Review by trygve-lie — CHANGES_REQUESTED** [review]
Can you elaborate a bit what this is?

Anyhow; I am afraid this can not live here. This package is to keep code which is reused between the different parts in Eik. Ex; validator methods we reuse between the client and the server. This will pull in Hapi as a dependency in all modules. I am not too keen on pulling in Hapi in the rollup import map plugin (which also depend on this module).

This seems like a very specific thing which perhaps should be in its separate module or something.

**Comment by gingermusketeer** [comment]
@trygve-lie hapi should be a development dependency not a normal dependency.  Have added this to the TODO for the PR.

This is building on the approach already present in the local-assets module which supports fastify and express at the moment. Don't have too much of an opinion as to whether this should be here or somewhere else.

**Comment by gingermusketeer** [comment]
This is difficult to test due to it being a nested dependency of the module being used. I think the test coverage is good enough that this should work as expected.

File: lib/helpers/local-assets.js [hunk 1]
```diff
@@ -26,19 +26,35 @@ async function localAssets(app, rootEikDirectory = process.cwd()) {
...
-                res.set('content-type', mapping.source.contentType);
-                fs.createReadStream(mapping.source.absolute).pipe(res);
-            } else if (res.type) {
-                // fastify
-                res.header('Access-Control-Allow-Origin', '*');
-                res.type(mapping.source.contentType);
-                res.send(fs.createReadStream(mapping.source.absolute));
-            }
-        });
+        const assetPath = mapping.destination.url.pathname;
```
Comment (digitalsadhu) [inline lib/helpers/local-assets.js:29]: Can we call `assetPath`, `pathname` to line up with the whatwg URL spec? When I think path, I think local and when I think pathname I think URL.

**Review by digitalsadhu — CHANGES_REQUESTED** [review]
Looks good to me, just the one comment.

File: lib/helpers/local-assets.js (hunk 1 above)
Comment (gingermusketeer) [reply lib/helpers/local-assets.js]: Done. I squashed the commits and added the `feat:` prefix so that this will be reflected in the Changelog.

Comment (digitalsadhu) [reply lib/helpers/local-assets.js]: noice

**Review by digitalsadhu — APPROVED** [review]
👍

**Comment by digitalsadhu** [comment]
@trygve-lie I've approved this and will merge it in for now. Since the module itself existed in this repo before this PR, I think that's fine and we can discuss how/where it lives eg. extracting it out after.

**Comment by trygve-lie** [comment]
I am still not convinced this should be here. 

The sole purpose of the this common module is that the code here is reused in the server, client and plugins. This goes as I see it the opposite way where we put all http framework stuff into one module. I would much rather that this was moved out to separate modules.

**Comment by trygve-lie** [comment]
@digitalsadhu I'm not super happy this got merged before we got this discussed.

**Comment by digitalsadhu** [comment]
Sorry about that, I agree it can be moved out, its just a very minor change to a function that's already there so I don't see the harm of making this change here now, move it out in a following step.

**Comment by digitalsadhu** [comment]
Also, looks like the releases are not working correctly. Should have bumped v3.0.0-next.1 -> v3.1.0-next.1 or something I'd have thought?

**Comment by trygve-lie** [comment]
I think this is a bit of a result of that a good chunk of code was moved from the cli to here without really going through what was moved. We do need something which can be hooked into http frameworks but I would really like that as a separate module. I am very afraid that we now will start to get non Eik projects depending on this common module directly and then we'll start getting a round where this needs to be cleaned up.

**Comment by digitalsadhu** [comment]
Totally fair point. Since its already there though (prior to this PR), the only solution is to extract it out and I don't want to focus on that right at the moment as I need to get the CLI in order. The best solution is probably to create an issue to discuss and remind us to make the change?

**Pull request merged**